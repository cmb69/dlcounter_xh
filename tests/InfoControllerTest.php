<?php

namespace Dlcounter;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeRequest;
use Plib\FakeSystemChecker;
use Plib\View;

class InfoControllerTest extends TestCase
{
    public function testRendersInfo(): void
    {
        $sut = new InfoController(
            "./",
            new FakeSystemChecker(),
            new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["dlcounter"])
        );
        Approvals::verifyHtml($sut->defaultAction(new FakeRequest()));
    }
}
