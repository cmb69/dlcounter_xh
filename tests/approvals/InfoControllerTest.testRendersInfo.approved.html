
<h1>Dlcounter 1.0beta2</h1>
<div>
  <h2>Synopsis</h2>
  <pre>{{{dlcounter('FILENAME')}}}</pre>
</div>
<div>
  <h2>System check</h2>
  <p class="xh_fail">Checking that the PHP version is at least 7.1.0 … failure</p>
  <p class="xh_warning">Checking that the PHP extension 'fileinfo' is loaded … warning</p>
  <p class="xh_fail">Checking that the CMSimple_XH version is at least 1.7.0 … failure</p>
  <p class="xh_fail">Checking that the CMSimple_XH version is at least 1.2 … failure</p>
  <p class="xh_fail">Checking that the CMSimple_XH plugin 'jquery' is installed … failure</p>
  <p class="xh_warning">Checking that './config/' is writable … warning</p>
  <p class="xh_warning">Checking that './css/' is writable … warning</p>
  <p class="xh_warning">Checking that './languages/' is writable … warning</p>
</div>
